src/mscorlib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -129,6 +129,24 @@ private struct Entry {
             }
         }
 
+        public Dictionary(IEnumerable<KeyValuePair<TKey, TValue>> enumerable):
+            this(enumerable, EqualityComparer<TKey>.Default) { }
+
+        public Dictionary(IEnumerable<KeyValuePair<TKey, TValue>> enumerable, IEqualityComparer<TKey> comparer):
+            this(comparer)
+        {
+            if( enumerable == null) {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.enumerable);
+            }
+
+            foreach (var pair in enumerable) {
```

Comment:
Nit: `KeyValuePair<TKey, TValue>` should be used here instead of `var` to match corefx coding style.